include/caffe/parallel.hpp

```diff
@@ -94,6 +94,9 @@ class P2PSync : public GPUParams<Dtype>, public Solver<Dtype>::Callback,
   }
 
   void run(const vector<int>& gpus);
+  void prepare(const vector<int>& gpus,
```

Comment:
This should probably be `Prepare` according to the style guide and existing style. Since we already have `run`, however, I guess we can postpone that to a future commit.